lib/i18n/backend/base.rb

```diff
@@ -26,15 +26,31 @@ def store_translations(locale, data, options = EMPTY_HASH)
 
       def translate(locale, key, options = EMPTY_HASH)
         raise I18n::ArgumentError if (key.is_a?(String) || key.is_a?(Symbol)) && key.empty?
+        if key.is_a?(Array)
+          key.each do |k|
+            raise I18n::ArgumentError if (k.is_a?(String) || k.is_a?(Symbol)) && k.empty?
```

Comment:
What do you think about making this check a function? It's duplicated now and possibly could use a name to wrap up the concept.